### ISSUE TITLE
Project generated from app spec has bad codesigning config

### ISSUE THREAD

**Issue by watt** [CocoaPods/CocoaPods#9110]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Create a new pod with an app spec. Run `pod gen` to create a project for that app spec. Open in Xcode, and try to enable codesigning so that I can build & install to a device.

## What did you expect to happen?

I can enable pretty easily enable codesigning with my already-installed iOS Developer cert, without mucking in the build settings.

## What happened instead?

The settings in Xcode look like this:

<img width="909" alt="1 signing disabled" src="https://user-images.githubusercontent.com/100192/63192455-ffd24400-c01f-11e9-8fe8-e4e00182a7dd.png">

(Clicking "Enable Development Signing" turns on "Automatically manage signing" but doesn't set a cert.)

Go into build settings and delete the config lines shown here.

<img width="864" alt="2 lines to delete" src="https://user-images.githubusercontent.com/100192/63192690-8e46c580-c020-11e9-97a0-d1822cbf4ff8.png">

Now the signing set up looks like this, which is how I would have expected it to look fresh out of the generator.

<img width="904" alt="3 signing ready to enable" src="https://user-images.githubusercontent.com/100192/63192712-9868c400-c020-11e9-954a-529e0236ee05.png">

After picking a profile, my local iOS Developer cert is picked up automatically.

After making this change manually, these lines were deleted from the generated `Pods.xcodeproj/project.pbxproj`:

```diff
                46EB2E00000210 /* Release */ = {
                        isa = XCBuildConfiguration;
                        baseConfigurationReference = 46EB2E00000310 /* Foo.example.xcconfig */;
                        buildSettings = {
                                ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
                                CODE_SIGNING_ALLOWED = YES;
                                CODE_SIGNING_REQUIRED = YES;
                                CODE_SIGN_IDENTITY = "iPhone Developer";
-                               "CODE_SIGN_IDENTITY[sdk=appletvos*]" = "";
-                               "CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "";
-                               "CODE_SIGN_IDENTITY[sdk=watchos*]" = "";
                                CURRENT_PROJECT_VERSION = 1;
                                GCC_PREFIX_HEADER = "Target Support Files/Foo/Foo-Example-prefix.pch";
                                INFOPLIST_FILE = "Example/Foo-Example-Info.plist";
                                IPHONEOS_DEPLOYMENT_TARGET = 9.0;
                                LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
                                PRODUCT_BUNDLE_IDENTIFIER = "org.cocoapods.${PRODUCT_NAME:rfc1034identifier}";
                                PRODUCT_NAME = "Foo-Example";
                                SDKROOT = iphoneos;
                                SWIFT_ACTIVE_COMPILATION_CONDITIONS = "$(inherited) ";
                                SWIFT_VERSION = 4.0;
                                TARGETED_DEVICE_FAMILY = "1,2";
                                VALIDATE_PRODUCT = YES;
                        };
                        name = Release;
                };
                46EB2E00000220 /* Debug */ = {
                        isa = XCBuildConfiguration;
                        baseConfigurationReference = 46EB2E00000310 /* Foo.example.xcconfig */;
                        buildSettings = {
                                ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
                                CODE_SIGNING_ALLOWED = YES;
                                CODE_SIGNING_REQUIRED = YES;
                                CODE_SIGN_IDENTITY = "iPhone Developer";
-                               "CODE_SIGN_IDENTITY[sdk=appletvos*]" = "";
-                               "CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "";
-                               "CODE_SIGN_IDENTITY[sdk=watchos*]" = "";
                                CURRENT_PROJECT_VERSION = 1;
                                GCC_PREFIX_HEADER = "Target Support Files/Foo/Foo-Example-prefix.pch";
                                INFOPLIST_FILE = "Example/Foo-Example-Info.plist";
                                IPHONEOS_DEPLOYMENT_TARGET = 9.0;
                                LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
                                PRODUCT_BUNDLE_IDENTIFIER = "org.cocoapods.${PRODUCT_NAME:rfc1034identifier}";
                                PRODUCT_NAME = "Foo-Example";
                                SDKROOT = iphoneos;
                                SWIFT_ACTIVE_COMPILATION_CONDITIONS = "$(inherited) ";
                                SWIFT_VERSION = 4.0;
                                TARGETED_DEVICE_FAMILY = "1,2";
                        };
                        name = Debug;
                };
```

It seems like this could be fixed by having the project generator could just omit these lines?

### Stack

```
   CocoaPods : 1.8.0.beta.1
        Ruby : ruby 2.6.1p33 (2019-01-30 revision 66950) [x86_64-darwin18]
    RubyGems : 3.0.1
        Host : Mac OS X 10.14.6 (18G87)
       Xcode : 10.2.1 (10E1001)
         Git : git version 2.20.1
Ruby lib dir : /Users/awatt/.rvm/rubies/ruby-2.6.1/lib
Repositories : master - git - https://github.com/CocoaPods/Specs.git @ e09fa43ade029febbdb61491210d061b23dab15c
               [redacted]
               trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /Users/awatt/.rvm/gems/ruby-2.6.1/bin/pod
```

### Plugins

```
claide-plugins                        : 0.9.2
cocoapods-amicable                    : 0.3.0
cocoapods-atlas                       : 0.4.0
cocoapods-binary                      : 0.4.3
cocoapods-check                       : 1.0.2
cocoapods-deintegrate                 : 1.0.4
cocoapods-dependencies                : 1.2.0
cocoapods-disable-podfile-validations : 0.1.1
cocoapods-generate                    : 1.5.0
cocoapods-mix-frameworks              : 0.1.0
cocoapods-packager                    : 1.5.0
cocoapods-packing-cubes               : 0.4.0
cocoapods-plugins                     : 1.0.0
cocoapods-search                      : 1.0.0
cocoapods-sq-command-metrics          : 0.5.0
cocoapods-stats                       : 1.1.0
cocoapods-trunk                       : 1.3.1
cocoapods-try                         : 1.1.0
cocoapods_debug                       : 0.1.0
refinement                            : 0.2.2
```

## Project that demonstrates the issue

The sample project I uploaded to #9105 demonstrates this: [Foo.zip](https://github.com/CocoaPods/CocoaPods/files/3499562/Foo.zip)

**Comment by dnkoutso** [comment]
Will fix for 1.8.0.beta.2

**Comment by mrcperello** [comment]
This issue is still present in 1.8.3
